R/install_torsten.R

```diff
@@ -0,0 +1,197 @@
+#'   release candidate to install. See <https://github.com/metrumresearchgroup/Torsten/releases>.
+#'   The URL should point to the tarball (`.tar.gz.` file) itself, e.g.,
+#'   `release_url="https://github.com/metrumresearchgroup/Torsten/archive/refs/tags/torsten_v0.90.0.tar.gz"`.
+#'   If both `version` and `release_url` are specified then `version` will be used.
+#' @param cpp_options (list) Any makefile flags/variables to be written to
+#'   the `make/local` file. For example, `list("CXX" = "clang++")` will force
+#'   the use of clang for compilation.
+#' @param check_toolchain (logical) Should `install_torsten()` attempt to check
+#'   that the required toolchain is installed and properly configured. The
+#'   default is `TRUE`.
+#'
+#' @examples
+#' \dontrun{
+#' # install_torsten(cores = 4)
+#'
+#' cpp_options <- list(
+#'   "CXX" = "clang++",
+#'   "CXXFLAGS+= -march=native",
+#'   PRECOMPILED_HEADERS = TRUE
+#' )
+#' # install_torsten(cores = 4, cpp_options = cpp_options)
+#' }
+#'
+
+#'## install_torsten version without unexported dependencies
+
+latest_torsten_release <- function() {
+  require(jsonlite)
+  dest_file <- tempfile(pattern = "releases-", fileext = ".json")
+  download_url <- "https://api.github.com/repos/metrumresearchgroup/Torsten/releases/latest"
+  release_list_downloaded <- utils::download.file(url = download_url,
+                                                  destfile = dest_file,
+                                                  quiet = TRUE)
+  if (release_list_downloaded != 0) {
+    stop("GitHub download of release list failed.", call. = FALSE)
+  }
+  release <- jsonlite::read_json(dest_file)
+  release$tag_name
+}
+
+get_torsten_release_list <- function() {
+  require(jsonlite)
+  dest_file <- tempfile(pattern = "releases-", fileext = ".json")
+  download_url <- "https://api.github.com/repos/metrumresearchgroup/Torsten/releases"
+  release_list_downloaded <- utils::download.file(url = download_url,
+                                                  destfile = dest_file,
+                                                  quiet = TRUE)
+  if (release_list_downloaded != 0) {
+    stop("GitHub download of release list failed.", call. = FALSE)
+  }
+  release <- jsonlite::read_json(dest_file)
+  sapply(release,
+         function(x){
+           x$tag_name
+         })
+}
+
+install_torsten <- function(dir = NULL,
+                            cores = getOption("mc.cores", 2),
+                            quiet = FALSE,
+                            timeout = 1200,
+                            version = NULL,
+                            release_url = NULL,
+                            cpp_options = list(),
+                            check_toolchain = TRUE
+) {
+
+  ## install_torsten is based on cmdstanr::install_cmdstan. Differences include
+  ## * install_torsten always overwrites pre-existing Torsten installation at the
+  ##.  specified cmdstan_path.
```

Comment:
By default, I don't think `install_torsten` should overwrite an existing directory. I'm in favor of adding an `overwrite` argument and setting it to `FALSE` be default.

Reply:
I implemented `overwrite`.